.husky/pre-commit

```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
```

Comment:
we should also add prettier and ts-compile

Also I think that we have to add --max-warnings 0 flag

Reply:
LintStaged runs eslint and prettier. What is ts-compile?

Reply:
Sryy, my bad, it's just compiling the ts code using tsconfig.json

`"ts-compile": "tsc -p tsconfig.json --noEmit",`

Sry, I'm used to different semantics

Reply:
yeeeeah... but build takes time, even on powerful machines (+5 seconds to time that lint-staged takes)
and who is dumb enough to push their code without compiling it first?
i don't think we need to do that really
in our last line of defense there is CI that WILL fail if there are type errors

Reply:
Yeah, that adds some time to precommit (with all commans that sould be I guess about ~ 20s)
But that's good practice, especially that on branch we would have clean history if we would like to go back or smth, each commit would be working as a charm

In case of minor changes we can always run commit with --no-verify flag

Reply:
20 SECONDS? no please let's not do this
instead I suggest to introduce corporal punishment for every time ur pipeline fails

Reply:
it is going to be a hell for person like me who likes to make a lots of commits